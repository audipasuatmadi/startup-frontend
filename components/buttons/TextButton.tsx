import React from 'react';
import BaseButton from './BaseButton';

export interface TextButtonProps {
  name?: string
  disabled?: boolean;
  href?: string;
  children?: any;

  className?: string;
  type?: 'button' | 'submit' | 'reset';

  onClick?: (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
  onMouseDown?: (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void;
  theme?: 'brand' | 'gray';
}

const TextButton: React.FC<TextButtonProps> = ({
  className,
  theme,
  disabled,
  ...otherProps
}: TextButtonProps) => {
  const themeList: [string, string][] = [];
  themeList['brand'] = ['brand', 'brand-light'];
  themeList['gray'] = ['gray-500', 'gray-300'];

  const [bgColor, bgLighterColor] = themeList[theme];

  const disabledFalseTheme = `text-${bgColor} transition-colors duration-200 hover:bg-opacity-30 hover:bg-${bgLighterColor} active:bg-${bgColor} focus:ring-4 ring-${bgColor} ring-opacity-50`;
  const disabledTrueTheme = `text-gray-400 cursor-default`;

  const finalStyles = `button bg-none ${
    disabled ? disabledTrueTheme : disabledFalseTheme
  } `;

  return (
    <BaseButton
      className={`${finalStyles}${className ? ` ${className}` : ''}`}
      disabled={disabled}
      {...otherProps}
    />
  );
};

TextButton.defaultProps = {
  theme: 'brand',
};

export default TextButton;
