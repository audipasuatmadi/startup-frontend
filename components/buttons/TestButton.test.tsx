import React from 'react'
import TextButton from './TextButton';
import { ShallowWrapper, shallow } from 'enzyme';

const dummyText = 'dummy text';

describe('text button rendering tests', () => {
  let wrapper: ShallowWrapper;

  const renderComponent = (props?: {}) => <TextButton {...props} />;
  afterEach(() => {
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('renders correctly without props', () => {
    wrapper = shallow(renderComponent());
  });

  it('renders correctly as a Link', () => {
    wrapper = shallow(renderComponent({ to: '/' }));
  });

  it('renders correctly with a children', () => {
    wrapper = shallow(renderComponent({ children: dummyText }));
  });

  it('renders correctly as a Link with a children', () => {
    wrapper = shallow(renderComponent({ to: '/', children: dummyText }));
  });

  it('renders correctly when disabled', () => {
    wrapper = shallow(renderComponent({ disabled: true }));
  });

  it('renders correctly as a Link when disabled', () => {
    wrapper = shallow(renderComponent({ to: '/', disabled: true }));
  });
});