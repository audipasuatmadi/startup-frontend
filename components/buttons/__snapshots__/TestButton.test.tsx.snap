// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text button rendering tests renders correctly as a Link 1`] = `
<BaseButton
  className="button bg-none text-brand transition-colors duration-200 hover:bg-opacity-30 hover:bg-brand-light active:bg-brand focus:ring-4 ring-brand ring-opacity-50 "
  to="/"
/>
`;

exports[`text button rendering tests renders correctly as a Link when disabled 1`] = `
<BaseButton
  className="button bg-none text-gray-400 cursor-default "
  disabled={true}
  to="/"
/>
`;

exports[`text button rendering tests renders correctly as a Link with a children 1`] = `
<BaseButton
  className="button bg-none text-brand transition-colors duration-200 hover:bg-opacity-30 hover:bg-brand-light active:bg-brand focus:ring-4 ring-brand ring-opacity-50 "
  to="/"
>
  dummy text
</BaseButton>
`;

exports[`text button rendering tests renders correctly when disabled 1`] = `
<BaseButton
  className="button bg-none text-gray-400 cursor-default "
  disabled={true}
/>
`;

exports[`text button rendering tests renders correctly with a children 1`] = `
<BaseButton
  className="button bg-none text-brand transition-colors duration-200 hover:bg-opacity-30 hover:bg-brand-light active:bg-brand focus:ring-4 ring-brand ring-opacity-50 "
>
  dummy text
</BaseButton>
`;

exports[`text button rendering tests renders correctly without props 1`] = `
<BaseButton
  className="button bg-none text-brand transition-colors duration-200 hover:bg-opacity-30 hover:bg-brand-light active:bg-brand focus:ring-4 ring-brand ring-opacity-50 "
/>
`;
